astropy/table/table.py

```diff
@@ -335,7 +335,10 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
             data = None
 
         if isinstance(data, self.Row):
-            data = data._table[data._index:data._index + 1]
+            if data._index == -1:
```

Comment:
I would suggest to do this as:
```
# Ensure that the index is positive
index = data._index % len(data._table)
data = data._table[index:index + 1]
```
For any valid `data._index`, taking mod(table length) will make the index positive.

Reply:
@taldcroft - It seems to me that `_index` should not be negative in the first place. Wouldn't it be better to just ensure `0 <= index < n` in `Row.__init__`? We're testing that `index` is within range there anyway.

p.s. @nksaunders - thanks for the fix!

Reply:
Great point, the fix should indeed be upstream in the Row init.

Reply:
I moved the change upstream to `Row.__init__()`, and set `self._index = self._index % n` to ensure it's positive. This is a simpler fix and still works as expected for other negative indices (i.e. `-2`). Thanks for the suggestions @taldcroft and @mhvk!